dataprofiler/data_readers/csv_data.py

```diff
@@ -86,6 +86,22 @@ def delimiter(self):
     def header(self):
         return self._header
 
+    def _check_and_return_options(self, options):
+        """
+        Ensures options are valid inputs to the data reader.
+
+        :param options: dictionary of options for the csv reader to validate
+        :type options: dict
+        :return: None
+        """
+        options = super()._check_and_return_options(options)
+        for key, value in options.items():
+            if key == 'header' and value != 'auto' and value is not None \
+                    and not (isinstance(value, int) and value > -1):
+                raise ValueError('`header` must either be \'auto\' or an '
+                                 'integer > -1')
```

Comment:
should we change to non-negative integer?

Reply:
+1, describing the header relative to the value -1 is strange

Reply:
Personally, I think this description is okay... but should probably give a full explanation. i.e.

"`auto`, `none` for no header, or a non-negative integer for the row that represents the header" 

Something like that.

Reply:
Sure, I find the visual values more direct but that's personal preference. Sometimes I feel blobs of text are less interpretable

Reply:
I think it just needs to be clear what the "header" number is. IMO saying `> -1` is fine, but it assumes they know.